src/util/taproot.rs

```diff
@@ -268,7 +268,7 @@ impl TaprootSpendInfo {
             internal_key: internal_key,
             merkle_root: merkle_root,
             output_key_parity: parity,
-            output_key: output_key,
+            output_key: TweakedPublicKey::new(output_key),
```

Comment:
This constructor should've been called `dangerous_assume_tweaked` :(

Reply:
Yup agreed. I was going to bring this in this review. Don't know how it skipped my review :( during the original one. 

Along with changing the new API to dangerous_assume_tweaked, we can also the logic here to that output_key is computed using the `taptweak` method instead of manually tweaking as implemented here. That way output_key would be of the required type directly.

Reply:
To implement these suggestions I added 4 additional patches to this PR. The first two of which are cleanup
- Remove comments: https://github.com/rust-bitcoin/rust-bitcoin/pull/728/commits/d84e3590a8436a4229bc7838b12415de9a9dcb74
- Use `debug_assert!` instead of `unreachable!`: https://github.com/rust-bitcoin/rust-bitcoin/pull/728/commits/8424aa13b22bcc78647adf69f7be19eaef1de596
- Implement @Kixunil suggestion, rename constructor: https://github.com/rust-bitcoin/rust-bitcoin/pull/728/commits/1f71b7506836da85c0875f1b0fda510c9ca2c278
- Implement @sanket1729 suggestion, use `tap_tweak` https://github.com/rust-bitcoin/rust-bitcoin/pull/728/commits/6f25d0ce186b48f684c81df9c73c8ba6cc5c95c6 but to do so I return the parity from `tap_tweak` also. This changes the taproot API, please review accordingly.